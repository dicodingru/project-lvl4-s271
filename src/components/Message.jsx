import React from 'react';
import PropTypes from 'prop-types';

const Message = (props) => {
  const {
    message: { username, text }
  } = props;
  return (
    <div className="border rounded mt-1 p-1">
      {username}: <span>{text}</span>
    </div>
  );
};

Message.propTypes = {
  message: PropTypes.shape({
    id: PropTypes.number,
    channelId: PropTypes.number,
    username: PropTypes.string,
    text: PropTypes.string
  }).isRequired
};

export default Message;
