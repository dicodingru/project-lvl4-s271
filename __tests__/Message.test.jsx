import React from 'react';
import renderer from 'react-test-renderer';
import Message from '../src/components/Message';

describe('Message component', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Message text="message text" />);
  });

  test('should render successfully', () => {
    expect(component.exists()).toEqual(true);
  });

  test('should match snapshot', () => {
    component = renderer.create(<Message text="message text" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
