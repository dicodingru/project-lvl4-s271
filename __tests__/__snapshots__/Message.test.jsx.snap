// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should match snapshot 1`] = `
<div>
  <span>
    message text
  </span>
</div>
`;
